diff --git a/gstlal-spiir/configure.ac b/gstlal-spiir/configure.ac
index 2be1ef41c..f574f9d5c 100644
--- a/gstlal-spiir/configure.ac
+++ b/gstlal-spiir/configure.ac
@@ -306,7 +306,7 @@ AS_IF([test "x$with_cuda" != "xno"], [
 	], [
 		CLIBS="lib"
 	])
-	AC_SUBST([NVCC], ["${with_cuda}/bin/nvcc"])
+	AC_SUBST([NVCC], ["${with_cuda}/bin/nvcc $NVCCFLAGS"])
 	AC_SUBST([NVCC_CFLAGS], ["-I${with_cuda}/include"])
 	PKG_CHECK_MODULES([GLIB], [glib-2.0])
 	AC_SUBST([NVCC_GSTLAL_CFLAGS], [$GLIB_CFLAGS])
diff --git a/gstlal-spiir/gst/cuda/Makefile.am b/gstlal-spiir/gst/cuda/Makefile.am
index 76f3500c9..16a8dc261 100644
--- a/gstlal-spiir/gst/cuda/Makefile.am
+++ b/gstlal-spiir/gst/cuda/Makefile.am
@@ -33,10 +33,19 @@ libcuda_plugin_la_CFLAGS = $(AM_CFLAGS) $(GSL_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFL
 
 libcuda_plugin_la_LIBADD = $(ADD_LIBS)
 
-libcuda_plugin_la_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(NVCC_LIBS) $(CHEALPIX_LIBS) -lstdc++ 
+libcuda_plugin_la_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(XML_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(NVCC_LIBS) $(CHEALPIX_LIBS) -lstdc++ 
+
+libgstlalspiir_la_CFLAGS = $(AM_CFLAGS) $(XML_CFLAGS) $(gstreamer_CFLAGS) $(GSTLAL_SPIIR_CFLAGS)
+
+default:
+	ifeq ($(strip $(libgstlalspiir_la_CFLAGS)),)
+			comp_FLAG =
+	else
+			comp_FLAG = --compiler-options=\"$(libgstlalspiir_la_CFLAGS)\"
+	endif
 
 .cu.lo:
-	$(top_srcdir)/gnuscripts/cudalt.py $@ $(NVCC) $(NVCC_CFLAGS) $(DEFAULT_INCLUDES) $(NVCC_LAL_CFLAGS) $(NVCC_GSTLAL_CFLAGS) $(NVCC_gstreamer_CFLAGS) $(ADD_CFLAGS) --ptxas-options=-v -O0  -maxrregcount=0 -gencode arch=compute_70,code=compute_70 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_30,code=sm_30 --compiler-options=\"$(libgstlalspiir_la_CFLAGS)\" -c $<
+	$(top_srcdir)/gnuscripts/cudalt.py $@ $(NVCC) $(NVCC_CFLAGS) $(DEFAULT_INCLUDES) $(NVCC_LAL_CFLAGS) $(NVCC_GSTLAL_CFLAGS) $(NVCC_gstreamer_CFLAGS) $(ADD_CFLAGS) --ptxas-options=-v -O0  -maxrregcount=0 -gencode  arch=compute_75,code=sm_75 -gencode arch=compute_70,code=compute_70 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_35,code=sm_35 $(comp_FLAG) -c $<
 
 noinst_HEADERS = \
 	spiir/spiir_kernel.h \
@@ -58,15 +67,15 @@ gstlal_cohfar_calc_fap_SOURCES = \
 	cohfar/background_stats_utils.c \
 	cohfar/cohfar_calc_fap.c
 
-gstlal_cohfar_calc_fap_CFLAGS = $(AM_CFLAGS) $(GSL_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS) $(ADD_CFLAGS)
-gstlal_cohfar_calc_fap_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(ADD_LIBS)
+gstlal_cohfar_calc_fap_CFLAGS = $(AM_CFLAGS) $(GSL_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS) $(XML_CFLAGS) $(ADD_CFLAGS)
+gstlal_cohfar_calc_fap_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(XML_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(ADD_LIBS)
 
 gstlal_postcoh_gen_detrsp_map_SOURCES = \
 	postcoh/postcoh_detrsp_map.c
 
 gstlal_postcoh_gen_detrsp_map_CFLAGS = $(AM_CFLAGS) $(GSL_CFLAGS) $(LAL_CFLAGS) $(GSTLAL_CFLAGS) $(gstreamer_CFLAGS) $(AM_CPPFLAGS) $(NVCC_CFLAGS) $(CHEALPIX_CFLAGS) $(ADD_CFLAGS)
 
-gstlal_postcoh_gen_detrsp_map_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(NVCC_LIBS) $(CHEALPIX_LIBS) $(ADD_LIBS) 
+gstlal_postcoh_gen_detrsp_map_LDFLAGS = $(AM_LDFLAGS) $(GSL_LIBS) $(LAL_LIBS) $(GSTLAL_LIBS) $(gstreamer_LIBS) $(XML_LIBS) $(GSTLAL_PLUGIN_LDFLAGS) $(NVCC_LIBS) $(CHEALPIX_LIBS) $(ADD_LIBS) 
 
 
 bin_PROGRAMS = \
diff --git a/gstlal-spiir/gst/cuda/multiratespiir/multiratespiir_kernel.cu b/gstlal-spiir/gst/cuda/multiratespiir/multiratespiir_kernel.cu
index dc1afb53a..e0d5617cd 100644
--- a/gstlal-spiir/gst/cuda/multiratespiir/multiratespiir_kernel.cu
+++ b/gstlal-spiir/gst/cuda/multiratespiir/multiratespiir_kernel.cu
@@ -241,8 +241,8 @@ __global__ void
 
         // Inter-Warp Reduction
         for (int off = WARPSIZE >> 1; off > 0; off = off >> 1) {
-            fltrOutptReal += __shfl_down(fltrOutptReal, off, 2 * off);
-            fltrOutptImag += __shfl_down(fltrOutptImag, off, 2 * off);
+            fltrOutptReal += __shfl_down_sync(0xFFFFFFFF, fltrOutptReal, off, 2 * off);
+            fltrOutptImag += __shfl_down_sync(0xFFFFFFFF, fltrOutptImag, off, 2 * off);
         }
 
         if (tLane == 0) {
@@ -266,8 +266,8 @@ __global__ void
                                : 0;
 
             for (int off = WARPSIZE >> 1; off > 0; off = off >> 1) {
-                sum_real += __shfl_down(sum_real, off);
-                sum_imag += __shfl_down(sum_imag, off);
+                sum_real += __shfl_down_sync(0xFFFFFFFF, sum_real, off);
+                sum_imag += __shfl_down_sync(0xFFFFFFFF, sum_imag, off);
             }
 
             if (threadIdx.x == 0) {
@@ -360,8 +360,8 @@ __global__ void cuda_iir_filter_kernel_fine(COMPLEX_F *cudaA1,
 
             // Inter-CU Reduction
             for (int off = cu >> 1; off > 0; off = off >> 1) {
-                fltrOutptReal += __shfl(fltrOutptReal, tLane + off, 2 * off);
-                fltrOutptImag += __shfl(fltrOutptImag, tLane + off, 2 * off);
+                fltrOutptReal += __shfl_sync(0xFFFFFFFF, fltrOutptReal, tLane + off, 2 * off);
+                fltrOutptImag += __shfl_sync(0xFFFFFFFF, fltrOutptImag, tLane + off, 2 * off);
             }
 
             if (tIdL == 0) {
diff --git a/gstlal-spiir/gst/cuda/postcoh/postcoh_kernel.cu b/gstlal-spiir/gst/cuda/postcoh/postcoh_kernel.cu
index 4a4dc87a9..cb870617c 100644
--- a/gstlal-spiir/gst/cuda/postcoh/postcoh_kernel.cu
+++ b/gstlal-spiir/gst/cuda/postcoh/postcoh_kernel.cu
@@ -99,9 +99,9 @@ __global__ void ker_max_snglsnr(COMPLEX_F **snr, // INPUT: snr
         // inter-warp reduction to find the max snr among threads in the warp
         for (int j = WARP_SIZE / 2; j > 0; j = j >> 1) {
 
-            temp_snr_sq = __shfl(max_snr_sq, wIDl + j, 2 * j);
+            temp_snr_sq = __shfl_sync(0xFFFFFFFF, max_snr_sq, wIDl + j, 2 * j);
 
-            temp_template = __shfl(max_template, wIDl + j, 2 * j);
+            temp_template = __shfl_sync(0xFFFFFFFF, max_template, wIDl + j, 2 * j);
             max_template  = (max_template + temp_template)
                            + (max_template - temp_template)
                                * (2 * (max_snr_sq > temp_snr_sq) - 1);
@@ -503,10 +503,10 @@ __global__ void ker_coh_max_and_chisq_versatile(
         }
 
         for (i = WARP_SIZE / 2; i > 0; i = i >> 1) {
-            stat_tmp           = __shfl_xor(stat_max, i);
-            snr_tmp            = __shfl_xor(snr_max, i);
-            nullstream_max_tmp = __shfl_xor(nullstream_max, i);
-            sky_idx_tmp        = __shfl_xor(sky_idx, i);
+            stat_tmp           = __shfl_xor_sync(0xFFFFFFFF, stat_max, i);
+            snr_tmp            = __shfl_xor_sync(0xFFFFFFFF, snr_max, i);
+            nullstream_max_tmp = __shfl_xor_sync(0xFFFFFFFF, nullstream_max, i);
+            sky_idx_tmp        = __shfl_xor_sync(0xFFFFFFFF, sky_idx, i);
 
             if (stat_tmp > stat_max) {
                 stat_max       = stat_tmp;
@@ -592,14 +592,17 @@ __global__ void ker_coh_max_and_chisq_versatile(
                 laneChi2 += (data.re * data.re + data.im * data.im);
             }
             for (int k = WARP_SIZE >> 1; k > 0; k = k >> 1) {
-                laneChi2 += __shfl_xor(laneChi2, k, WARP_SIZE);
+                laneChi2 += __shfl_xor_sync(0xFFFFFFFF, laneChi2, k, WARP_SIZE);
             }
             if (srcLane == 0) { snr_shared[wID] = laneChi2; }
             __syncthreads();
+
+            unsigned mask = __ballot_sync(0xFFFFFFFF, threadIdx.x < wn);
+
             if (threadIdx.x < wn) {
                 laneChi2 = snr_shared[srcLane];
                 for (i = wn / 2; i > 0; i = i >> 1) {
-                    laneChi2 += __shfl_xor(laneChi2, i);
+                    laneChi2 += __shfl_xor_sync(mask, laneChi2, i);
                 }
                 if (srcLane == 0) {
                     chisq_cur = laneChi2 / autocorr_norm[j][tmplt_cur];
@@ -699,10 +702,10 @@ __global__ void ker_coh_max_and_chisq_versatile(
             }
 
             for (i = WARP_SIZE / 2; i > 0; i = i >> 1) {
-                stat_tmp           = __shfl_xor(stat_max, i);
-                snr_tmp            = __shfl_xor(snr_max, i);
-                nullstream_max_tmp = __shfl_xor(nullstream_max, i);
-                sky_idx_tmp        = __shfl_xor(sky_idx, i);
+                stat_tmp           = __shfl_xor_sync(0xFFFFFFFF, stat_max, i);
+                snr_tmp            = __shfl_xor_sync(0xFFFFFFFF, snr_max, i);
+                nullstream_max_tmp = __shfl_xor_sync(0xFFFFFFFF, nullstream_max, i);
+                sky_idx_tmp        = __shfl_xor_sync(0xFFFFFFFF, sky_idx, i);
 
                 if (stat_tmp > stat_max) {
                     stat_max       = stat_tmp;
@@ -789,7 +792,7 @@ __global__ void ker_coh_max_and_chisq_versatile(
                     laneChi2 += (data.re * data.re + data.im * data.im);
                 }
                 for (int k = WARP_SIZE >> 1; k > 0; k = k >> 1) {
-                    laneChi2 += __shfl_xor(laneChi2, k, WARP_SIZE);
+                    laneChi2 += __shfl_xor_sync(0xFFFFFFFF, laneChi2, k, WARP_SIZE);
                 }
 
                 if (srcLane == 0) {
diff --git a/gstlal-spiir/lib/Makefile.am b/gstlal-spiir/lib/Makefile.am
index 237434d00..ca7c67a9f 100644
--- a/gstlal-spiir/lib/Makefile.am
+++ b/gstlal-spiir/lib/Makefile.am
@@ -9,5 +9,5 @@ libgstlalspiir_la_SOURCES = \
 libgstlalspiir_la_CFLAGS = $(AM_CFLAGS) $(AM_CPPFLAGS) $(XML_CFLAGS) $(gstreamer_CFLAGS) $(GSTLAL_SPIIR_CFLAGS)
 libgstlalspiir_la_LIBADD = 
 
-libgstlalspiir_la_LDFLAGS = $(AM_LDFLAGS) $(XML_LIBS) $(gstreamer_LIBS)
+libgstlalspiir_la_LDFLAGS = $(AM_LDFLAGS) $(XML_LIBS) $(gstreamer_LIBS) $(GLIB_LIBS)
 
