diff --git a/gstlal-spiir/python/pipemodules/postcoh_finalsink.py b/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
index ff379269f..44ec46266 100644
--- a/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
+++ b/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
@@ -619,6 +619,9 @@ class FinalSink(object):
         self.thread_upload_skymap = None
 
     def __is_significant_trigger(self, postcoh_inspiral):
+        if postcoh_inspiral.cohsnr > 37:
+            return True
+
         if not self.gracedb_far_threshold:
             return False
 
