diff --git a/gstlal-spiir/python/pipemodules/postcoh_finalsink.py b/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
index 0424506cc..a409b5a2c 100644
--- a/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
+++ b/gstlal-spiir/python/pipemodules/postcoh_finalsink.py
@@ -672,6 +672,9 @@ class FinalSink(object):
         self.thread_upload_skymap = None
 
     def __is_significant_trigger(self, postcoh_inspiral):
+        if postcoh_inspiral.cohsnr > 37:
+            return True
+
         if not self.gracedb_far_threshold:
             return False
 
