--- Makefile	2019-12-12 13:24:08.448227206 +1100
+++ Makefile	2019-12-06 15:25:10.151096744 +1100
@@ -1439,7 +1439,7 @@
 frameCPP_python.stamp: hdr-time-stamp
 
 Z-Sample_Python_ver8-600000000-1.gwf: test_frameCPP.py _frameCPP.la
-	env PYTHONPATH=$(LDAS_DIRECTORY_BUILD_LIB_FRAMECPP)/swig/python/.libs:$(LDAS_DIRECTORY_BUILD_LIB_FRAMECPP)/swig/python \
+	env PYTHONPATH=$(LDAS_DIRECTORY_BUILD_LIB_FRAMECPP)/swig/python/.libs:$(LDAS_DIRECTORY_BUILD_LIB_FRAMECPP)/swig/python:${PYTHONPATH} \
 	$(OS_LIBRARY_PATH)=$(FRAMESPEC_LIB_PATHS):$(LDASTOOLSAL_LIBDIRS):${$(OS_LIBRARY_PATH)}	 \
 	$(PYTHON) $(srcdir)/test_frameCPP.py
